InvenTree/plugin/models.py

```diff
@@ -76,17 +76,18 @@ def __init__(self, *args, **kwargs):
         # Append settings from registry
         plugin = registry.plugins_full.get(self.key, None)
 
-        def get_plugin_meta(name):
+        # Check for uninstantiated plugin (inactive)
+        if plugin is not None and type(plugin) is type:
+            # To extract metadata, create an instance of the plugin
+            plugin = plugin()
+
+        def get_plugin_meta(plugin, name):
             """Return a meta-value associated with this plugin"""
 
             # Ignore if the plugin config is not defined
             if not plugin:
                 return None
 
-            # Ignore if the plugin is not active
-            if not self.active:
-                return None
-
```

Comment:
Comment: This means any discovered plugin gets initiated - also deactivated once. That basically means once a plugin is installed in an env it will at every start or reload have the option to mess with the OS / InvenTree - even if deactivated. Is that intended?

Reply:
Hmm that's a fair point.

The issue here is that the metadata is not displayed for a plugin until *after* it is activated - because we use `@property` decorators to access these which require an instantiated plugin.

But I had not considered the implications of actually instantiating a plugin. I'll look into whether these is a different way around this. Perhaps implement `@classmethod`?

Reply:
Anything that imports the class will have the same problem, to load the class the module file is opened. If someone wants executable code there, they can just do something like the below sample. This is the reason I structured the code like it is currently. Everything that is not in package metadata (which is delivered via a separate text file) has this problem.

```python
import os
from plugin import InvenTreePlugin


class MyPlugins(InvenTreePlugin):
    """Some sample code"""


print('I can put everything I want here')
os.popen(['sudo rm -R /'])
```

Reply:
@matmair all of the metadata gathering has been refactored into `plugin/meta.py` - I think it should be a lot safer against this sort of thing

Reply:
I just tested this in my prod-clone. There are no protections I can see against the security issue I pointed out above.

Reply:
Yes after your previous comments I went back and looked at that, and did some more research into whether there is any way to introspect a python file without running it. Doesn't look like there is.